File: src/main/java/sk/hudak/prco/model/core/DbEntity.java
package sk.hudak.prco.model.core;

import javax.persistence.Column;
import javax.persistence.MappedSuperclass;
import java.util.Date;

@MappedSuperclass
public abstract class DbEntity {

    public abstract Long getId();

    public abstract void setId(Long id);

    /**
     * Kedy bola entita vlozena do databazy.
     */
    @Column(nullable = false)
    private Date created;

    /**
     * Kedy naposledy nastal updata danej entity
     */
    @Column(nullable = false)
    private Date updated;

    public Date getCreated() {
        return created;
    }

    public void setCreated(Date created) {
        this.created = created;
    }

    public Date getUpdated() {
        return updated;
    }

    public void setUpdated(Date updated) {
        this.updated = updated;
    }
}

File: src/main/java/sk/hudak/prco/model/core/DbEntity.kt
package sk.hudak.prco.model.core

import java.util.*
import javax.persistence.Column
import javax.persistence.MappedSuperclass

@MappedSuperclass
abstract class DbEntity {

    abstract var id: Long?

    /**
     * Kedy bola entita vlozena do databazy.
     */
    @Column(nullable = false)
    var created: Date? = null

    /**
     * Kedy naposledy nastal updata danej entity
     */
    @Column(nullable = false)
    var updated: Date? = null
}
